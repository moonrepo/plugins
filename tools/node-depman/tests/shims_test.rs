#[cfg(not(windows))]
mod node_depman_tool {
    use proto_pdk_test_utils::*;

    mod npm {
        use super::*;

        generate_shims_test!("npm-test", ["npm", "npx", "node-gyp"]);
    }

    mod pnpm {
        use super::*;

        generate_shims_test!("pnpm-test", ["pnpm", "pnpx"]);
    }

    mod yarn {
        use super::*;

        generate_shims_test!("yarn-test", ["yarn", "yarnpkg"]);
    }
}
