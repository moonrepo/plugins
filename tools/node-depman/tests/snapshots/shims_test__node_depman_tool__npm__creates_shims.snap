---
source: tools/node-depman/tests/shims_test.rs
expression: "std :: fs ::\nread_to_string(sandbox.path().join(\".proto/shims/registry.json\")).unwrap()"
---
{
  "node-gyp": {
    "alt_bin": true,
    "parent": "npm-test"
  },
  "npm": {
    "alt_bin": true,
    "parent": "npm-test"
  },
  "npx": {
    "alt_bin": true,
    "parent": "npm-test"
  }
}
